modules/InboundEmail/InboundEmail.php

```diff
@@ -5302,7 +5302,41 @@ public function returnImportedEmail($msgNo, $uid, $forDisplay = false, $clean_em
                     return false;
                 }
                 $email->$rel->add($email->parent_id);
-            }
+            } else {
+                //Assign Parent Values if references header mentions a sent email in the system
+                if(!empty($header->references) || !empty($header->in_reply_to)) {
+                    $references = explode(" ",  $header->in_reply_to . " " . $header->references);
+                    foreach ($references as $reference) {
+                        $referenceId = rtrim(ltrim($reference, '<'), '>');
+                        if (strncmp($referenceId, "SUITECRM_", 9) === 0) {
+                            $query = 'SELECT id, assigned_user_id, parent_id, parent_type FROM emails WHERE emails.message_id = \'' . $referenceId . '\' and type = \'out\' and emails.deleted = 0 LIMIT 1';
+                            $results = $this->db->query($query, true);
+                            $row = $this->db->fetchByAssoc($results);
+                            if(!empty($row)) {
+                                $email->parent_id = $row['parent_id']; // set email parent to parent of referenced email
+                                $email->parent_type = $row['parent_type'];
+                                switch ($this->assignment_behavior) {
+                                    case 'donothing'; // Don't assign the email to a user
+                                        break;
+                                    case 'recordowner'; // Assign the email to the owner of the record the referenced email's parrent
+                                        $parentBean = BeanFactory::getBean($row['parent_type'], $row['parent_id']);
+                                        if (isset($parentBean->id)) {
```

Comment:
Testing for $parentBean->id but using $parentBean->assigned_user_id. Is it intend?